.ci/magician/cmd/collect_nightly_test_status.go

```diff
@@ -0,0 +1,213 @@
+	if err != nil {
+		return fmt.Errorf("Error getting GA nightly test status: %w", err)
+	}
+
+	err = createTestReport(provider.Beta, tc, gcs, formattedStartCut, formattedFinishCut, date)
+	if err != nil {
+		return fmt.Errorf("Error getting Beta nightly test status: %w", err)
+	}
+
+	return nil
+}
+
+func createTestReport(pVersion provider.Version, tc TeamcityClient, gcs CloudstorageClient, formattedStartCut, formattedFinishCut, date string) error {
+	// Get all service test builds
+	builds, err := tc.GetBuilds(pVersion.TeamCityNightlyProjectName(), formattedFinishCut, formattedStartCut)
+	if err != nil {
+		return err
+	}
+
+	var testInfoList []TestInfo
+	for _, build := range builds.Builds {
+		serviceTestResults, err := tc.GetTestResults(build)
+		if err != nil {
+			return fmt.Errorf("failed to get test results: %v", err)
+		}
+		if len(serviceTestResults.TestResults) == 0 {
+			continue
+		}
+		// Get service package name
+		serviceName, err := convertServiceName(build.BuildTypeId)
+		if err != nil {
+			return fmt.Errorf("failed to convert test service name for %s: %v", build.BuildTypeId, err)
+		}
+
+		for _, testResult := range serviceTestResults.TestResults {
+			var errorMessage string
+			// Get test debug log gcs link
+			logLink := fmt.Sprintf("https://storage.cloud.google.com/teamcity-logs/nightly/%s/%s/%s/debug-%s-%s-%s-%s.txt", pVersion.TeamCityNightlyProjectName(), date, build.Number, pVersion.ProviderName(), build.Number, strconv.Itoa(build.Id), testResult.Name)
+			// Get concise error message
+			if testResult.Status == "FAILURE" {
+				errorMessage = convertErrorMessage(testResult.ErrorMessage)
+			}
+			testInfoList = append(testInfoList, TestInfo{
+				Name:         testResult.Name,
+				Status:       testResult.Status,
+				Service:      serviceName,
+				ErrorMessage: errorMessage,
+				LogLink:      logLink,
+			})
+		}
+	}
+
+	// Write test status data to a JSON file
+	fmt.Println("Write test status")
+	testStatusFileName := fmt.Sprintf("%s-%s.json", date, pVersion.String())
+	err = utils.WriteToJson(testInfoList, testStatusFileName)
+	if err != nil {
+		return err
+	}
+
+	// Upload test status data file to gcs bucket
+	objectName := pVersion.String() + "/" + testStatusFileName
+	err = gcs.WriteToGCSBucket(NIGHTLY_DATA_BUCKET, objectName, testStatusFileName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// convertServiceName extracts service package name from teamcity build type id
+// input: TerraformProviders_GoogleCloud_GOOGLE_NIGHTLYTESTS_GOOGLE_PACKAGE_SECRETMANAGER
+// output: secretmanager
+func convertServiceName(servicePath string) (string, error) {
+	idx := strings.LastIndex(servicePath, "_")
+
+	if idx != -1 {
+		return strings.ToLower(servicePath[idx+1:]), nil
+	}
+	return "", fmt.Errorf("wrong service path format for %s", servicePath)
+}
+
+// convertErrorMessage returns concise error message
+func convertErrorMessage(rawErrorMessage string) string {
```

Comment:
Could you add some tests for this?

In particular I think that if startMarker is not found it might cause errors.

Reply:
Good catch! I've modified the logic and added unit tests. Thanks!